sycl/include/CL/sycl/buffer.hpp

```diff
@@ -134,13 +134,12 @@ class buffer {
     if (b.is_sub_buffer())
       throw cl::sycl::invalid_object_error(
           "Cannot create sub buffer from sub buffer.");
-    // TODO: SYCL CTS are failed because not appropriate offset and range
-    // are passed to this constructor which causes a throw of exception.
-    // Some changes need to be done there.
-    // if (isOutOfBounds(baseIndex, subRange, b.Range))
-    //   throw cl::sycl::invalid_object_error("Out-of-bounds size");
-    // if (!isContiguousRegion(baseIndex, subRange, b.Range))
-    //   throw cl::sycl::invalid_object_error("Non-contiguous region");
+    if (isOutOfBounds(baseIndex, subRange, b.Range))
+      throw cl::sycl::invalid_object_error(
+          "Requested sub-buffer size exceeds the size of the parent buffer");
+    if (!isContiguousRegion(baseIndex, subRange, b.Range))
+      throw cl::sycl::invalid_object_error(
+          "Requested sub-buffer region is not continuous");
```

Comment:
```suggestion
          "Requested sub-buffer region is not contiguous");
```
Not sure if the spelling is intentionally continuous in this case.

Reply:
Continuous - forming an unbroken whole; without interruption.
Contiguous - sharing a common border; touching.

I am not an English expert, but in my opinion continuous is more suitable here since the memory region represented by sub buffer must be without any interruptions.

Reply:
Contiguous is the accepted term for this.

Reply:
Fixed the "typo"

Reply:
Probably "sharing a common border; touching" is to be understood as each element touches its neighbors.